src/types/request.rs

```diff
@@ -14,6 +14,7 @@ pub struct Request {
     #[pyo3(from_py_with = "get_body_from_pyobject")]
     pub body: Vec<u8>,
     pub url: Url,
+    pub ip_addr: String,
```

Comment:
Just thinking, would an optional string make more sense here? Rather than giving them an empty string if it's missing? Just a thought. Would also save allocating if we didn't have one

Reply:
That's actually a good idea, how would that translate on the Python side? as just a `None`?

Reply:
@IdoKendo , yes. A `None | str`